Bartending+ PR, Shaking and Stirring for Drinks.

AlexisMoonlight opened the pull request:
<!--
This is a semi-strict format, you can add/remove sections as needed but the order/format should be kept the same
Remove these comments before submitting
-->

# Description

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

Merge [Bartending+ PR by Vermidia](https://github.com/space-wizards/space-station-14/pull/29243), adding shaking and stirring to most of upstream drink recipes. 
Add shaking and stirring to most of floof drinks.
Add a drink, Pussy Tonic.

---

# TODO
[ ] DeltaV drinks do not have shaking and stirring, as they have not merged Bartending+ PR. It is possible to add them to floof versions. I do not know if that would be a good idea, so I have not done it yet.
<!--
A list of everything you have to do before this PR is "complete"
You probably won't have to complete everything before merging but it's good to leave future references
-->

---

<!--
This is default collapsed, readers click to expand it and see all your media
The PR media section can get very large at times, so this is a good way to keep it clean
The title is written using HTML tags
The title must be within the <summary> tags or you won't see it
-->

<details><summary><h1>Media</h1></summary>
<p>

<!-- EDITORS NOTE: please make sure there's two blank lines surrounding each attachment, otherwise github may refuse to render it properly. I fixed it for you. -->

![drinks+ shake](https://github.com/user-attachments/assets/452f47af-3c61-46d7-a261-4c3b2589335e)

![drinks+ stir](https://github.com/user-attachments/assets/3b280d26-82fc-4b02-ba92-28addaf7067c)

</p>
</details>

---

# Changelog

<!--
You can add an author after the `:cl:` to change the name that appears in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl:
- add: Added Bartending+ PR by Vermidia, shaking and stirring to most of drinks.
- add: Addes shaking and stirring to most of Floof drinks.
- add: Added Pussy Tonic drink

AlexisMoonlight committed:
Floof drinks modified

AlexisMoonlight committed:
Drinks+ merge

AlexisMoonlight committed:
PussyTonic drink

Mnemotechnician commented:
Sounds amazing. But please link any PRs you've cherry-picked.

AlexisMoonlight commented:
Sure, updated the head, but also [here it is](https://github.com/space-wizards/space-station-14/pull/29243)

Mnemotechnician commented:
Well, the test fail is valid, you have a duplicate CreamOfCoconut prototype for some reason.

AlexisMoonlight commented:
Seems it originally was in upstream drinks, but now a different one is in floof prototypes for some reason. Removed upstream one, testing.

AlexisMoonlight committed:
Cream of coconut duplicate removed

ShadesMars reviewed (dismissed):
Test fail seem valid:

https://github.com/Fansana/floofstation1/actions/runs/15310442131/job/43077017729?pr=1039

AlexisMoonlight committed:
Replace SolutionContainerSystem with SharedSolutionContainerSystem

AlexisMoonlight committed:
Remove whitespace

AlexisMoonlight commented:
I'll be honest, I have no idea what caused it to fail. The local instance works as intended, Content.Tests passes with one skip and it seems like my laptop cannot run Content.IntegrationTests, so I can't do much of my own testing. I have replaced one outdated system in the PR, maybe that'll help, though I doubt it.

Memeji commented:
I'm rerunning the test to double check...

The issue seems to be from "CafeDeCacao" as that was the last one testing before the fail.
It might not be finding the mixture, or there is a similar one that exists so it has trouble figuring out which one to make.
I'll look into it if I have time this week, but I would check if the mixture of reagents turn into only CafeDeCacao or can it turn into something else as well?

AlexisMoonlight commented:
Removed the floof-exclusive drinks yml - still fails.
Removed upstream drinks yml - passes.
Saw your comment and tried removing CafeDeCacao from upstream yml - fails, this time the last drink is DirtyWastelander
So the problem is in upstream drinks yml, now to figure out where exactly.

AlexisMoonlight commented:
Ok, removing BOTH CafeDeCacao and DirtyWastelander seems to have fixed it.

AlexisMoonlight committed:
Removed requiredMixerCategories from CafeDeCacao and DirtyWastelander

Memeji commented:
Odd, I don't see anything obvious like a conflicting recipe.

AlexisMoonlight commented:
So, it works. Although this kinda brings up an issue. The tests failed because 2 of 3 parts of the recipes reacted with each other before they could be stirred. Removing stirring fixes it, sure, but if you add the parts in wrong order in the game, they will still react with each other. But unfortunately that would need to be PRed to upstream first, I believe.

Memeji commented:
Just need a second review from @FoxxoTrystan or @Mnemotechnician then we'll be good to go.

Mostly to be sure regarding:

> So, it works. Although this kinda brings up an issue. The tests failed because 2 of 3 parts of the recipes reacted with each other before they could be stirred. Removing stirring fixes it, sure, but if you add the parts in wrong order in the game, they will still react with each other. But unfortunately that would need to be PRed to upstream first, I believe.

Memeji commented:
@Mnemotechnician @Fansana @FoxxoTrystan
When ya get the chance to go over this PR, let me know about the last message above.

AlexisMoonlight commented:
Wanna possibly clear something up, just in case. The changes work, I've tested them in a local setting, both shaking and stirring made recipes that require them complete. I've not tested every drink changed, which is why we had those test fails as 2 recipes contained 3 reagents, 2 of which could also react with each other. Adding shaking/stirring to those recipes made them impossible as the two reagents would react with each other before the doafter could complete. I've removed stirring from them, so they are now working the same as before the PR, where adding ingredients in a wrong order can mess up the drink because of the two ingredients that can react with each other. I could also change them, but they are from an upstream recipe, so changing them here I believe to be a wrong way to do it, I'll probably try to do a PR to upstream to fix that eventually.
If I have to test manually in local every drink and provide screenshots for this PR to be merged I'll do it, just please say so, bacause I thought test passes are enough for that. (While I cannot run full integration test suite locally, I can run individual tests, and I did specifically run TryAllReactions and it worked locally too)

Mnemotechnician reviewed (approved):
Answering your question, no, you don't have to manually test every recipe. The tests are passing and one of them is basically a "try to complete every recipe" kind of test, which is more than enough.

AlexisMoonlight commented:
Yeah, that's what I figured, just was getting antsy, I guess.

Mnemotechnician commented:
Going to go ahead and merge this now, sorry for the delay.
